networkx/algorithms/isomorphism/tests/test_isomorphism.py

```diff
@@ -41,15 +49,17 @@ def test_could_be_isomorphic(self):
         assert iso.could_be_isomorphic(self.G3, self.G2)
         assert not iso.could_be_isomorphic(self.G1, self.G6)
 
-    def test_fast_could_be_isomorphic(self):
-        assert iso.fast_could_be_isomorphic(self.G3, self.G2)
-        assert not iso.fast_could_be_isomorphic(self.G3, self.G5)
-        assert not iso.fast_could_be_isomorphic(self.G1, self.G6)
+    @pytest.mark.parametrize("fn", (iso.fast_could_be_isomorphic, fast_cbi))
```

Comment:
Can we add a test for the example where the full table is needed (not each row separately) to report that they are not isomorphic?  (suggestions put at beginning of previous change -- but no change occurs to that line.)  
```suggestion
    def test_full table_of_properties(self):
        G1 = nx.Graph([(0, 1), (0, 2), (0, 3), (0, 4), (1, 2), (3, 4), (4, 5), (4, 6)])
        G2 = G1.copy()
        G1.add_edge(0, 7)
        G2.add_edge(4, 7)
        assert not nx.could_be_isomorphic(G1, G2)

    @pytest.mark.parametrize("fn", (iso.fast_could_be_isomorphic, fast_cbi))
```

Reply:
Yes we definitely should - I tried but failed to come up with such a case :)

Unfortunately this proposed case also doesn't work because `G1.order()` = 7 and `G2.order() = 8` (order is the very first check)

Reply:
Hmmm.... I think both G1 and G2 have nodes `0`-`7` so they should have `order=8` ...  right?
I tried drawing an ascii picture for this.
```
G1: 1   3    G2: 1   3       
    |\ / \       |\ / \     
    | 0---4      | 0---4   
    |/ \  |\     |/   /|\  
    2   7 6 5    2   7 6 5  
```
The key being `1, 2, 3, 5, 6, 7` don't change. `0` and `4` have two and one triangles in both. respectively. but their degree gets switched between 4 and 5. So the degree sequence is the same, the triangle sequence is the same, but if you compare (degree, triangle) pairs G1 has a `[(5,2), (4,1)]` while `G2` has `[(4,2), (5,1)]`.  This one doesn't require the cliques. But it does require making the property tuples for each node.

Reply:
You're right, I screwed up the G1 and G2 (i.e. I did `G1.add_edge` twice :) ).

On a related note, I decided to investigate potential test cases by brute force! Specifically, looking at all fully-connected, non-isomorphic pairs of graphs with 7 nodes (drawn from the graph atlas). There are a couple hundred such examples :)

Reply:
Okay - I added the test in 62c825c . It's a nice test that probes areas that the current tests don't, but unfortunately it doesn't quite get all the way to exactly what we want - the reason for this is the difference between sorting the properties individually vs. combined! It turns out that considering the combined degree-triangle property is enough to establish a difference, and the clique comparison is never really needed.

I think this test should be added and we should continue trying to find others! I have potential cases from the brute-force search over all 7-node connected graphs that I will investigate further!

Reply:
I believe I've reached the bottom of the rabbit hole...

According to my analysis (which I intend to propose as an nx-guide, because I found it quite interesting & fun), there are only two possible graph-pairs out of all fully-connected graphs with 7 nodes that have the same *combined* degree-triangle distribution but a *different* maximal-clique distribution. In other words, using the terminology proposed in this PR, there are only two graph pairs that satisfy `could_be_isomorphic(G, H, properties="dt")` but don't satisfy `could_be_isomorphic(G, H, properties="dtc")`:

![pair1](https://github.com/user-attachments/assets/3a301280-a569-4a1d-ac29-25539b345532)

![pair2](https://github.com/user-attachments/assets/fa905118-d553-4ac1-9494-ebc31e4a7909)

Reply:
Nice work! 
I spent a while looking for pairs of graphs that had the same d and t and c sequences but didn't have the same sequence of 3-tuples. I learned a lot about how maximal clique counts and degree interact (and had good fun)! But the graphs started getting pretty big.  I like these examples a lot! :)